### DISCUSSION TITLE
Copy & paste not working between Win10 guest to host or host to Win10 guest

### DISCUSSION THREAD

**Discussion by theglove44** [utmapp/UTM#5867] [Windows]
**Describe the issue**  
Issue with no copy & paste function between host to client, or client to host. I've searched the help documents and made sure that the guest tools are installed and that that clipboard sharing is enabled in the settings.

My guest machine is a Window 10 machine for reference as well.

**Configuration**  
* UTM Version: 4.4.4 (92)
* macOS Version: Sonoma 14.0
* Mac Chip (Intel, M1, ...): Intel

**Crash log**  
N/A

**Debug log**  
N/A

**Upload VM**  
[config.plist.zip](https://github.com/utmapp/UTM/files/13164370/config.plist.zip)

**Comment by osy** [comment]
Try reinstalling the guest tools and restarting.

**Reply by theglove44** [reply]
I think it's something Windows specific. As a test I've just loaded up a new Ubuntu instance and the copy paste works fine across that.

I did re-install the guest tools in the Windows VM and reboot but I still had the same issue. Keyboard short cuts to copy and paste didn't work and the right click option was greyed out.

**Comment by plia7** [comment]
Did you figure it out? I have the same issue with Windows 7

**Comment by andriusr** [comment]
Similar issue between UTM 4.6.4 and MacOS 15.4.1. Guest OS Win 11.

Copy/paste from Mac to Win11 works fine.
From Win 11 to MacOS it does not work for most of the time. I have noticed that if I copy something from Notepad to MacOS, it works only for short strings, like 20 characters or so. If string is longer, then it is not pasted into MacOS at all.

**Comment by allthetime** [comment]
Also having this issue from Guest Fedora Linux -> MacOS Host. Mac -> Linux works fine.

**Reply by mikew** [reply]
Same with Arch ARM on macOS: Host -> Guest works, but Guest -> Host doesn't. `spice-vdagent` is running, and `wl-clipboard` is installed.

**Reply by tux-linux** [reply]
I have the same issue.
